MapboxNavigation/Style.swift

```diff
@@ -209,6 +210,98 @@ public class ResumeButton: UIControl {
     }
 }
 
+/// :nodoc:
+@objc(MBSpeedLimitSign)
+open class SpeedLimitSign: UIView {
+    
+    private static let textInsets = UIEdgeInsets(top: 6, left: 6, bottom: 6, right: 6)
+    
+    lazy var label: SpeedLimitLabel = .forAutoLayout()
+    
+    let defaultText = NSLocalizedString("SPEED_LIMIT", bundle: .mapboxNavigation, value: "Speed\nLimit\n", comment: "Speed limit sign main text.")
+    let paragraphStyle = NSMutableParagraphStyle()
+    
+    var userIsOverSpeedLimit = false
+    
+    var speedLimit: SpeedLimit? {
+        didSet {
+            guard let speedLimit = speedLimit else { return }
+            let fontSize = UIFont.systemFontSize
+            
+            let strings = NSMutableAttributedString()
+            
+            strings.append(NSMutableAttributedString(string: defaultText.uppercased(), attributes: [
+                .paragraphStyle: paragraphStyle,
+                .font: UIFont.systemFont(ofSize: fontSize * 0.9)
+                ]))
+            
+            strings.append(NSMutableAttributedString(string: String(describing: speedLimit.speed), attributes: [
+                .font: UIFont.boldSystemFont(ofSize: fontSize * 2),
+                .paragraphStyle: paragraphStyle,
+                .foregroundColor: userIsOverSpeedLimit ? #colorLiteral(red:0.91, green:0.20, blue:0.25, alpha:1.0) : UIColor.black
```

Comment:
Do we actually want to change the color when over the speed limit?

Reply:
I think we should. I don't know if there's a general rule but in Sweden, it's a 10% tolerance on the speed limit. Changing to red when you're going 61 on a 60 road could be too aggressive but maybe 66 including the 10% tolerance could work?

Reply:
This could also be a ramp that goes goes from black to red exponentially.

Reply:
Yeah I'm thinking about moving this into a followup pr.

Reply:
Tolerance varies by jurisdiction. In some localities in the U.S., the tolerance is 0&nbsp;mph; in others, it’s 5&nbsp;mph or higher. I don’t think the threshold is something we should determine on the client side. Moreover, the Directions API doesn’t yet distinguish advisory speed limits from legal speed limits.